File: frontend/src/types/simulation.ts
export interface SimulationRunRequest {
  vehicle_count: number;
  events_per_vehicle: number;
  seed?: number;
  start_time?: string;
}

export interface SimulationRunResponse {
  status: string;
  vehicles_created: number;
  plates_created: number;
  events_created: number;
  seed?: number;
  start_time: string;
  end_time: string;
  duration_seconds: number;
  message: string;
}

export interface SimulationStatusResponse {
  is_running: boolean;
  total_simulated_vehicles: number;
  total_simulated_events: number;
  total_simulated_plates: number;
  last_run_seed?: number;
  last_simulation_time?: string;
}

export interface SimulationCleanupResponse {
  status: string;
  detections_deleted: number;
  plates_deleted: number;
  vehicles_deleted: number;
  message: string;
}

File: backend/app/schemas/simulation.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field, ConfigDict


class SimulationRunRequest(BaseModel):
    vehicle_count: int = Field(default=30, ge=1, le=200, description="Number of synthetic vehicles to simulate")
    events_per_vehicle: int = Field(default=5, ge=1, le=20, description="Target number of camera sightings per vehicle")
    seed: Optional[int] = Field(default=42, description="Random seed for deterministic reproducibility")
    start_time: Optional[datetime] = Field(default=None, description="Starting reference timestamp (defaults to recent UTC)")


class SimulationRunResponse(BaseModel):
    status: str = "completed"
    vehicles_created: int
    plates_created: int
    events_created: int
    seed: Optional[int]
    start_time: datetime
    end_time: datetime
    duration_seconds: float
    message: str


class SimulationStatusResponse(BaseModel):
    is_running: bool = False
    total_simulated_vehicles: int
    total_simulated_events: int
    total_simulated_plates: int
    last_run_seed: Optional[int] = None
    last_simulation_time: Optional[datetime] = None


class SimulationCleanupResponse(BaseModel):
    status: str = "cleaned"
    detections_deleted: int
    plates_deleted: int
    vehicles_deleted: int
    message: str
